pkg/controller/cluster/cluster.go

```diff
@@ -368,16 +373,21 @@ func (c *ClusterReconciler) bindNodeProxyClusterRole(ctx context.Context, cluste
 	return c.Client.Update(ctx, clusterRoleBinding)
 }
 
-func (c *ClusterReconciler) agent(ctx context.Context, cluster *v1alpha1.Cluster, serviceIP, token string) error {
-	agent := agent.New(cluster, serviceIP, c.SharedAgentImage, c.SharedAgentImagePullPolicy, token)
-	agentsConfig := agent.Config()
-	agentResources, err := agent.Resources()
-	if err != nil {
-		return err
+type ResourceEnsurer interface {
```

Comment:
can we add this method to the agent interface thats already exists in agent.go

Reply:
I removed that interface, but I can move this in the `agent` package if you prefer. 👍

Reply:
yeah we can move it to the agent, the reason I did that interface so that we can add to it later if needed, but I think its fine for now